plugins/imjournal/imjournal.c

```diff
@@ -520,24 +520,32 @@ persistJournalState(void)
 	DEFiRet;
 	FILE *sf = NULL; /* state file */
 	char tmp_sf[MAXFNAME];
-	size_t n;
 
 	DBGPRINTF("Persisting journal position, cursor: %s, at head? %d\n",
 			  journalContext.cursor, journalContext.atHead);
 
 	/* first check that we have valid cursor */
 	if (!journalContext.cursor) {
+		DBGPRINTF("Journal cursor is not valid, ok...\n");
 		ABORT_FINALIZE(RS_RET_OK);
 	}
 
 	/* we create a temporary name by adding a ".tmp"
 	 * suffix to the end of our state file's name
+	 *
+	 * we use snprintf() to safely honor the boundaries
+	 * of the temporary state file name buffer by using
+	 * a precision specifier, which will limit the number
+	 * of bytes taken from cs.stateFile to what will fit
+	 *
+	 * TODO: figure out a better way to avoid the PATH_MAX
+	 * problem. The truncated cs.stateFile with .tmp at the
+	 * end is not optimal
 	 */
-	n = snprintf(tmp_sf, sizeof(tmp_sf), "%s.tmp", cs.stateFile);
-	if (n >= sizeof(tmp_sf)) { /* backup in case of too long path name */
-		strncpy(tmp_sf, cs.stateFile, sizeof(tmp_sf) - 5);
-		strcat(tmp_sf, ".tmp");
-	}
+#define IM_SF_TMP_SUFFIX ".tmp"
+	snprintf(tmp_sf, sizeof(tmp_sf), "%.*s%s",
+			(int)(sizeof(tmp_sf) - strlen(IM_SF_TMP_SUFFIX) -1),
```

Comment:
I suggest
```
(int)(sizeof(tmp_sf) - sizeof(IM_SF_TMP_SUFFIX) -1),
```
which I guess was the original intent (sizeof vs. strlen). With strlen I think the remainung buffer space may be one byte too few.

Reply:
I believe it is clearer to leave it with `strlen()` since that is the intent - to reserve space for the length of the suffix and one for the null, and also because it would remain correct even if `IM_SF_TMP_SUFFIX` were changed to be declared as `char *`. Using -1 is correct to reserve space for the null byte in this case. I think `- sizeof() - 1` would reserve too much space.

Reply:
OK. Just two words of warning

1. not all compilers optimize out `strlen("constant")`, so this can lead to bigger and slower
   code
1. the rest of the code uses `(sizeof("constant")-1)`, so this would be the only occasion
    where `strlen("constant")` is used - which may cause some confusion (like mine on the meaning 
   of  `-1`)

I would still accept this, because it is imjournal, which is special in many ways and slow in any case. If it were some other component, I'd insist on `sizeof()` ;-)